bin/servicerouter.py

```diff
@@ -577,13 +687,44 @@ def config(apps, groups):
                 r'[^a-zA-Z0-9\-]', '_',
                 backendServer.host + '_' + str(backendServer.port))
 
+            healthCheckOptions = None
+            if app.healthCheck:
+                server_health_check_options = None
+                if app.mode == 'tcp':
+                    server_health_check_options = templater \
+                        .haproxy_backend_server_tcp_healthcheck_options
+                elif app.mode == 'http':
+                    server_health_check_options = templater \
+                        .haproxy_backend_server_http_healthcheck_options
+                if server_health_check_options:
+                    healthCheckOptions = server_health_check_options.format(
+                        healthCheck=app.healthCheck,
+                        healthCheckPortIndex=app.healthCheck['portIndex'],
+                        healthCheckProtocol=app.healthCheck['protocol'],
+                        healthCheckPath=app.healthCheck['path'],
+                        healthCheckTimeoutSeconds=app.healthCheck[
+                            'timeoutSeconds'],
+                        healthCheckIntervalSeconds=app.healthCheck[
+                            'intervalSeconds'],
+                        healthCheckIgnoreHttp1xx=app.healthCheck[
+                            'ignoreHttp1xx'],
+                        healthCheckGracePeriodSeconds=app.healthCheck[
+                            'gracePeriodSeconds'],
+                        healthCheckMaxConsecutiveFailures=app.healthCheck[
+                            'maxConsecutiveFailures'],
+                        healthCheckFalls=app.healthCheck[
+                            'maxConsecutiveFailures'] + 1
+                    )
+
             backend_server_options = templater.haproxy_backend_server_options
             backends += backend_server_options.format(
                 host=backendServer.host,
                 port=backendServer.port,
                 serverName=serverName,
                 cookieOptions=' check cookie ' +
-                serverName if app.sticky else ''
+                serverName if app.sticky else '',
+                healthCheckOptions=' ' +
```

Comment:
Why do we need to add the space here? Could we just default healthCheckOptions to empty string (or empty string) and get rid of this if?